node/hyperbahn/index.js

```diff
@@ -358,6 +358,19 @@ function advertise(opts) {
     }
 };
 
+HyperbahnClient.prototype.unadvertise =
+function unadvertise() {
+    var self = this;
+    if (self.state !== States.ADVERTISED) {
+        return;
```

Comment:
you dont want to early return; always close the timer.

Just because `state === UNADVERTISED` does not mean that the timer is gone. because it could be trying in a loop